### ISSUE TITLE
Pylance Intellisense not functioning

### ISSUE THREAD

**Issue by AzuStar** [microsoft/pylance-release#1486]
## Environment data
-   Language Server version: 2021.6.2 (pyright dafa497d)
-   OS and version: Win10 1909 build 18363.778
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.2 and 3.9.5 (tested both)
-   VSCode Latest as of 23/6/2021

## Expected behaviour

helloworld.py example
literally following the guide here, that's what is expected (autocompletion, suggestions etc)
https://code.visualstudio.com/docs/python/python-tutorial#_prerequisites

DataExtractor.py example
![image](https://user-images.githubusercontent.com/52426335/123022580-e5acde00-d3cd-11eb-8a41-8b008af186f1.png)
Not only Jedi suggests .moderator()
It also resolves type and shows correct documentation.

## Actual behaviour

Default clean installation of vscode.
Opened workspace.
New file -> "helloworld.py"
![image](https://user-images.githubusercontent.com/52426335/123021748-91552e80-d3cc-11eb-8a63-c4a6d4eaf163.png)

As basic as it get, doesnt even suggest linspace()

But the real problem begins when it gets praw.
Again, same clean installation.
Opened workspace.
New file -> "DataExtractor.py"
![image](https://user-images.githubusercontent.com/52426335/123021974-f14bd500-d3cc-11eb-887a-30e859cfbf13.png)

Pylance just panics and quits. Pylance is still running, but it barely understand/suggest anything.

## Logs

Logs output is useless, but here:
```
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
Background analysis message: setFileOpened
Background analysis message: markFilesDirty
[FG] completion at \vm\random\c9h3c84n-nf12oo1q\extracted-data\DataExtractor.py:16:42 ...
[FG]   parsing: \vm\random\c9h3c84n-nf12oo1q\extracted-data\DataExtractor.py (3ms)
[FG]   binding: \vm\random\c9h3c84n-nf12oo1q\extracted-data\DataExtractor.py (1ms)
[FG] completion at \vm\random\c9h3c84n-nf12oo1q\extracted-data\DataExtractor.py:16:42 [found 0 items] (53ms)
[FG] completion at \vm\random\c9h3c84n-nf12oo1q\extracted-data\DataExtractor.py:16:42 [found 0 items] (0ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
```

## Code Snippet / Additional information

In short, expected behaviour is completely achieved, when extension Pylance is uninstalled, but it is obviously not expected behaviour because then Jedi becomes a language server :)
Jedi would be fine if it wasn't dumb :kekw:

**Comment by jakebailey** [comment]
This should have been fixed in more recent versions of numpy. What version of numpy are you using? Can you upgrade and see if your issue is fixed?

> Pylance just panics and quits.

What do you mean by this? Just that things are missing, or that it crashed in some way?

That reddit library may need some stubs to help describe how its API works.

**Comment by AzuStar** [comment]
Numpy appears to be outdated. Updating to the latest, fixed the issue.
As for PRAW, there are quite a lot of stubs that describe everything in detail. Jedi even picks up 90% of them, but Pylance doesn't :/
As for "panics and quits" my bad entirely. Should have described better, but no, it's perfectly fine and running, just doesn't display anything.

**Comment by jakebailey** [comment]
> As for PRAW, there are quite a lot of stubs that describe everything in detail.

If this library ships stubs, we should be picking them up. Installing `praw`, I don't see stubs, but they do have some inline types. For your specific case, it seems like the subreddit helper's `__call__` annotation relies on `models` existing on `praw`, which it is not declared to be:

![image](https://user-images.githubusercontent.com/5341706/123840659-73536680-d8c3-11eb-880c-fadc65f3a41f.png)

![image](https://user-images.githubusercontent.com/5341706/123840663-764e5700-d8c3-11eb-90af-756272e83847.png)

Perhaps the praw authors would be interested in testing things out and using https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md as a guide to how to more correctly set up their types.

**Comment by jakebailey** [comment]
Adding `import .models as models` to `__init__.py` fixes things:

![image](https://user-images.githubusercontent.com/5341706/123840896-c7f6e180-d8c3-11eb-868f-702c722dd2b0.png)

**Comment by AzuStar** [comment]
Well, if it was specifically 1 library, then sure, why not. But this is happening quite often, which is still annoying.
Another example is TextBlob.
To reproduce, install latest TextBlob
```python
from textblob import TextBlob
txtBlob = TextBlob("Some text. Test me.")
for sentence in txtBlob.sentences:
   print(sentence)
```
txtBlob.sentence raises error during analysis
`
"cached_property" is not iterable
  "__iter__" method not defined
`

Simple dictionary, cannot resolve types as well, which could be the cause of populating the dictionary with Unknowns.
However, I populate dictionary by calling json.load() and from there I pretty much working blind without suggestions T.T

Thanks for your replies! Your suggestions did help btw. Much love <3

**Comment by jakebailey** [comment]
Those are type checking errors, not analysis errors. If you've enabled a type checking mode above the default ("off", i.e. no type checking), then you're not guaranteed to have a good experience when using libraries that aren't well typed.

**Comment by AzuStar** [comment]
I do have it enabled, but doesn't seem to be working.
I have pulled some older version of textblob and it seems to be working.
